File: Reverse_integer(7).py
class Solution(object):
    def reverse(self, x):
        if x < 0:
            n = -x       
            re = 0
            while n > 0:
                d = n % 10
                re = re * 10 + d
                n //= 10   
            re = -re   
        else: 
            n = x
            re = 0
            while n > 0:
                d = n % 10
                re = re * 10 + d
                n //= 10
        
        if re < -2**31 or re > 2**31 - 1:
            return 0
        return re
        

File: Reverse_integer(7).cpp
class Solution {
public:
    int reverse(int x) {
        long long rem = 0;
        if (x > 0) {
            long long val = x;
            while (val > 0) {
                int v = val % 10;
                rem = rem * 10 + v;
                val = val / 10;
            }
        } else {
            long val = -(long long)x;
            while (val > 0) {
                int v = val % 10;
                rem = rem * 10 + v;
                val = val / 10;
            }
            rem = -rem;
        }

        if (rem > INT_MAX || rem < INT_MIN) {
            return 0;
        }
        return (int)rem;
    }
};